packages/ariakit-utils/src/__tests__/dom-test.ts

```diff
@@ -0,0 +1,601 @@
+import {
+  closest,
+  contains,
+  getActiveElement,
+  getDocument,
+  getPopupRole,
+  getScrollingElement,
+  getTextboxSelection,
+  getWindow,
+  isButton,
+  isFrame,
+  isPartiallyHidden,
+  isTextField,
+  isVisible,
+  matches,
+  scrollIntoViewIfNeeded,
+} from "../dom";
+
+test("getDocument", () => {
+  const initialInnerHTML = document.body.innerHTML;
+  document.body.innerHTML = '<div id="testNode"></div>';
+  const node = document.getElementById("testNode");
+
+  expect(getDocument(undefined)).toEqual(document);
+  expect(getDocument(null)).toEqual(document);
+  expect(getDocument(node)).toEqual(document);
+
+  // Reset
+  document.body.innerHTML = initialInnerHTML;
+});
```

Comment:
Should we do the html cleanup using `beforeEach`/`afterEach`? Something like this?
```jsx
let initialInnerHTML;

beforeEach(() => {
  initialInnerHTML = document.body.innerHTML;
});

afterEach(() => {
  document.body.innerHTML = initialInnerHTML;
});
```